crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -539,6 +546,62 @@ System stdlib will not be used for module definitions.",
     }
 }
 
+/// Different kinds of conda environment
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub(crate) enum CondaEnvironmentKind {
+    /// The base Conda environment; treated like a system Python environment.
+    Base,
+    /// Any other Conda environment; treated like a virtual environment.
+    Child,
+}
+
+impl CondaEnvironmentKind {
+    /// Compute the kind of `CONDA_PREFIX` we have.
+    ///
+    /// When the base environment is used, `CONDA_DEFAULT_ENV` will be set to a name, i.e., `base` or
+    /// `root` which does not match the prefix, e.g. `/usr/local` instead of
+    /// `/usr/local/conda/envs/<name>`.
+    fn from_prefix_path(system: &dyn System, path: &SystemPath) -> Self {
+        // If we cannot read `CONDA_DEFAULT_ENV`, there's no way to know if the base environment
+        let Ok(default_env) = system.env_var(EnvVars::CONDA_DEFAULT_ENV) else {
+            return CondaEnvironmentKind::Child;
+        };
+
+        // These are the expected names for the base environment
+        if default_env != "base" && default_env != "root" {
+            return CondaEnvironmentKind::Child;
+        }
```

Comment:
N.B. this isn't discussed in your commentary but we only allow these names. I don't know if people use other default environment names, but there's never been a complaint in uv.

Reply:
This seems straightforward to change if necessary